### DISCUSSION TITLE
parseJSON / JSONparse documentation

### DISCUSSION THREAD

**Discussion by JMLX42** [Budibase/budibase#3724] [Help]
Hi,

I'm trying to do the following automation:

- Step 1: Create some JSON string (using the Bash node for example).
- Step 2: Parse that JSON to get one field as the value for a column of a Create Row

Here is the Bash code for step 1:

```bash
openssl genrsa -out private.pem 2048
openssl rsa -pubout -in private.pem > public.pem

jq --null-input \
	--rawfile private private.pem \
	--rawfile public public.pem \
	'{"private_key": $private, "public_key": $public}'
```

And the Create Row for step 2:

![image](https://user-images.githubusercontent.com/998703/145257364-af8cd39a-24d0-49fe-a7cc-60ae2e6b8254.png)

Instead of the expected result, the raw handlebar template is used as the value to create the row:

![image](https://user-images.githubusercontent.com/998703/145257595-82fae5ad-1ab4-4de4-b225-49272ea67ffe.png)

Am I missing something or should I file a new bug issue?

Regards,

**Comment by mjashanks** [comment]
This is the same issue as https://github.com/Budibase/budibase/issues/3344

I see that you commented in there already. We really need to fix this one...

**Reply by JMLX42** [reply]
> This is the same issue as #3344

@mjashanks IMHO not it is not related: it works fine if I use something else than `parseJSON` or `JSONparse`. For example the following works: `{{ steps.1.stdout }}`. So AFAIK this is not related to #3344.

The problem here is strictly related to `parseJSON` and `JSONparse`.

**Reply by mjashanks** [reply]
Ah, the reason that I though it's related to that was because there is no binding icon on the input - thus suggesting that it's just storing static text.

![image](https://user-images.githubusercontent.com/3524181/145446283-8476674c-c3ac-4967-937a-d4c8c66b6485.png)

I can see your issue now. Not sure why `jsonParse` is not working there. You could add a Javascript block before, and do

```javascript
return JSON.parse(steps.1.stdout);
```

**Reply by JMLX42** [reply]
> I can see your issue now. Not sure why jsonParse is not working there. You could add a Javascript block before, and do

@mjashanks yes but then how can I use that JSON in a Create Row?

**Reply by mjashanks** [reply]
so, say 
- `steps.1.stdout` is `{ "name" : "mike" }`
- Step 2 is your javascript action (as above)

then...

`{{ steps.2.result.name }}` would equal `mike`.

**Reply by JMLX42** [reply]
OK so I had two JS Scripting steps:

```javascript
return JSON.parse(steps[1].stdout).private_key
```

```javascript
return JSON.parse(steps[1].stdout).public_key
```

which I replaced with a single step:

```javascript
return JSON.parse(steps[1].stdout)
```

The output is:

```json
{
  "success": true,
  "value": {
    "private_key": "...",
    "public_key": "...",
  }
}
```

and then:

![image](https://user-images.githubusercontent.com/998703/145616830-3df42cdc-4f33-4fb5-b821-ea1bce3b0e6e.png)

but it does not work:

![image](https://user-images.githubusercontent.com/998703/145616889-07713525-f19b-4a4c-b05b-2b16805dd2ff.png)

**Reply by JMLX42** [reply]
As the output JSON shows a `value` field (instead of a `result` field as suggested [here](https://github.com/Budibase/budibase/discussions/3724#discussioncomment-1787054)), I tried:

![image](https://user-images.githubusercontent.com/998703/145617338-d41cd146-8b81-4d67-bfd9-bb78c2a5a5ca.png)

but I have the exact same error.

**Reply by JMLX42** [reply]
OK so the proper bindings are on `steps.2.result.value`:

![image](https://user-images.githubusercontent.com/998703/145617538-f5cbab01-4f6e-4772-af95-bd41aa395719.png)

@mjashanks is that expected?

(I still have this error though but it's unrelated: https://github.com/Budibase/budibase/discussions/3710)
